### PR TITLE
Reorg Part I: Data

### PR DESCRIPTION
**Pull request by samet-akcay**
## Description

- Part I of this reorg is to re-structure the data subpackage.

### CODE REVIEW

File: src/anomalib/utils/types/task_type.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 """Task type enum."""
 
-# Copyright (C) 2022 Intel Corporation
+# Copyright (C) 2022-2023 Intel Corporation
```
Comment (ashwinvaidya17) [inline src/anomalib/utils/types/task_type.py:3]:
```suggestion
# Copyright (C) 2023 Intel Corporation
```

**Review by ashwinvaidya17 — COMMENTED** [review]
Thanks. Just one minor comment

File: src/anomalib/utils/types/task_type.py (hunk 1 above)
Comment (samet-akcay) [reply src/anomalib/utils/types/task_type.py:3]: as far as I know this is the suggested way when we create the file earlier and modify it later on.

Comment (ashwinvaidya17) [reply src/anomalib/utils/types/task_type.py:3]: Ah didn't know. Let's follow this convention from now on then.

**Review by ashwinvaidya17 — APPROVED** [review]

**Pull request merged**